### PR TITLE
mold: drop 1.3.1, add package_type

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/mold/all/conanfile.py [hunk 1]
```diff
@@ -93,17 +99,21 @@ def package(self):
...
         self.cpp_info.resdirs = []
 
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs.extend(["m", "pthread", "dl"])
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs = ["m", "pthread", "dl"]
+
+        bindir = os.path.join(self.package_folder, "bin")
+        mold_executable = os.path.join(bindir, "mold")
+        self.conf_info.define("user.mold:mold", mold_executable)
```
Comment (uilianries) [inline recipes/mold/all/conanfile.py:112]:
```suggestion
        self.conf_info.define("user.mold:path", mold_executable)
```

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**